### ISSUE TITLE
Add a way for users to edit and create Topics

### ISSUE THREAD

**Issue by MattGiesa** [CivicTechTO/women-and-color-frontend#43]
## Design
![image](https://user-images.githubusercontent.com/23535291/36826753-a396201c-1cdd-11e8-87fa-86f29d9ff9af.png)
[Link to Invision](https://projects.invisionapp.com/share/YJCPM4TQ6#/screens/265986247)

## Description
I can't find a place to enter topics, so I'm assuming it still needs to be created. The designs show a neat autocomplete + list of [chips](https://material-ui-next.com/demos/chips/) on the "CMS-About" page which seems to map to our "Tell us about you" (Profile.js) page. I'm unsure if topics exist in the backend at all. I'm also unsure if the Profile.js page needs to be updated to match the layout in the designs - this should be a discussion in a separate issue.

Endpoint for topics: `/api/v1/topics`

Ex. Search topics by query string: `/api/v1/topics/?q=<query>`
Ex. Create new topic: POST to `/api/v1/topics` with the json body `{ "topic": "my topic name" }`

## Work Needed
- [ ] [Link Backend Issue](https://github.com/CivicTechTO/women-and-color-frontend/issues/000)
- [ ] Works on mobile
- [ ] Works on desktop
- [ ] Wrote some tests
- [ ] (Some Functionality that is expected)

**Comment by s-kennedy** [comment]
Estimated hours: 3

**Comment by s-kennedy** [comment]
Here's an option for the dropdown compontent: https://github.com/JedWatson/react-select

**Comment by kidcompassion** [comment]
Hi folks - I don’t know if this helps that much, given that I’m a React rookie and will likely take awhile to do it, but I’m happy to take a shot at this over the week/weekend (and FWIW if I’m holding anything up, no one feel bad about taking it away from me :joy: :joy: :joy:). This is what I’m assuming I’ll have to do to complete this task:

- [ ] Create a route in the front end for a user’s profile editing page (are routes already mapped out somewhere, or do I just pick something temporary, like /profile/edit/?)
- [ ] Create a blank admin page at that route
- [ ] Create this new topics form component using react-select (or something similar), which I can put in my shiny new admin page
- [ ] Wire the component up to the api endpoint so my topics are getting saved as expected

Am I in the ballpark, @s-kennedy ?

**Comment by dragonza** [comment]
Could you explain more about the two inputs of topics. The first one will show all the topics that we have in the systems. I am not sure about the second input though.

**Comment by kidcompassion** [comment]
@dragonza I interpreted the graphic as showing one input with two states; the first state being when the autocomplete is expanded, and the second when it is not.

**Comment by s-kennedy** [comment]
@kidcompassion @dragonza Thanks for picking up this issue!

The route for the edit profile page is already set here: https://github.com/CivicTechTO/women-and-color-frontend/blob/master/app/config/routes.js

There are sub-routes for the About and Account pages, which you can find here: https://github.com/CivicTechTO/women-and-color-frontend/blob/master/app/pages/EditProfile/EditProfile.js

So you can just add the topics selector to the `/profile/about` page, which is here: https://github.com/CivicTechTO/women-and-color-frontend/blob/master/app/pages/EditProfile/About/About.js

Don't worry about making a custom dropdown component to look exactly like the mockup, the react-select package already has the functionality we need and it looks clean, so let's just go with that. You just need to:

- [ ] add a Topics section on the About page
- [ ] add the react-select component and the help text below it
- [ ] use the AsyncCreatable higher-order component from the react-select package, as documented here: https://github.com/JedWatson/react-select#combining-async-and-creatable
- [ ] hook up the topics API endpoint to the `loadOptions` function
- [ ] save new option to database in `onNewOptionClick` function

**Comment by dragonza** [comment]
Hi I try to create a new topics
![image](https://user-images.githubusercontent.com/10185439/38717014-4fbcc1b4-3f0e-11e8-9fb4-bf30d4e893ba.png)
but I receive this message. Am I missing something here?

**Comment by s-kennedy** [comment]
@dragonza the request body should be JSON, not text. You can change it in the dropdown in Postman.

**Comment by dragonza** [comment]
Oh thank you very much!

**Comment by kidcompassion** [comment]
Hi folks! 

I've been chipping away at this this weekend, but I think it's over my current skill level. I've successfully imported the desired component, but I've run up against an issue where React-Select, by default, imports its component as "Select", which is colliding with the material-ui select naming: 

<img width="1154" alt="screen shot 2018-04-15 at 10 41 53 am" src="https://user-images.githubusercontent.com/1507827/38780913-aa6ce034-409a-11e8-8547-446f576a55a5.png">

I set up a workaround option of building a custom component with the same functionality, but my lack of skill with Redux has me stuck.

My work to date is available in **feature/43-topics-selector**, and if anyone has time to nudge me along I'm happy to finish, but given that this is tagged as blocked, I'm also okay with it if someone else wants to take over due to time constraints. Am un-assigning myself in the meantime, just so it's clear the ticket is available.
